mongo/gridfs/download_stream.go

```diff
@@ -235,6 +235,11 @@ func (ds *DownloadStream) fillBuffer(ctx context.Context) error {
 			_ = ds.cursor.Close(ctx)
 			return ds.cursor.Err()
 		}
+		// If there are no more chunks, but we didn't read the expected number of chunks, return an
+		// ErrWrongIndex error to indicate that we're missing chunks at the end of the file.
+		if ds.expectedChunk != ds.numChunks {
+			return ErrWrongIndex
```

Comment:
What's the difference between `ErrWrongIndex` and `ErrWrongSize`?

Reply:
Our documentation says:

```
// ErrWrongIndex is used when the chunk retrieved from the server does not have the expected index.
...
// ErrWrongSize is used when the chunk retrieved from the server does not have the expected size.
```

`ErrWrongIndex` seems more accurate in this situation than `ErrWrongSize`. I could also see an argument for a new error type, something like `ErrMissingChunks`.

Reply:
I agree that the error name/description is kinda confusing and an `ErrMissingChunk` would be more clear. However, since it's caused by the same issue where `ErrWrongIndex` is currently used (i.e. a chunk is missing), I decided it was better to use the same error so any existing error checking code would continue working.

I originally decided to use `ErrWrongIndex` because the unified spec test describing the error it's expecting as ["ChunkIsMissing"](https://github.com/mongodb/specifications/blob/d268119c9d4354f60e59d2e798c3cb3ee5323a29/source/gridfs/tests/download.yml#L176-L190). In the Go Driver, there isn't a "ChunkIsMissing" error, but the current spec test runner maps ["ChunkIsMissing" to `ErrWrongIndex`](https://github.com/mongodb/mongo-go-driver/blob/d2ec1fc6d615aa3822f7350706f73307c8df433a/mongo/integration/gridfs_spec_test.go#L486-L487).

I think we should use `ErrWrongIndex` in both cases, but maybe we can/should update the error message? Something like:
> chunk is missing or index does not match expected index

Reply:
Using `ErrWrongIndex` SGTM, but I'm hesitant to change the message for `ErrWrongIndex` in case users are checking for the current message. That said, GridFS does not get much usage, and most users would probably do something like `errors.Is(err, gridfs.ErrWrongIndex)`, anyway, so up to you.

Reply:
I'll leave it as-is and add a proposal to the Go Driver 2.0 scope to rename and/or update the description of the error.